fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java

```diff
@@ -140,30 +148,45 @@ public String create(final String apiRequestBodyAsJson) {
         return this.toApiJsonSerializer.serialize(result);
     }
 
-    @PUT
-    @Path("{userId}")
-    public String update(@PathParam("userId") final Long userId, final String apiRequestBodyAsJson) {
-
-        final CommandWrapper commandRequest = new CommandWrapperBuilder() //
-                .updateUser(userId) //
-                .withJson(apiRequestBodyAsJson) //
-                .build();
-
-        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-
-        return this.toApiJsonSerializer.serialize(result);
+  @PUT
+  @Path("/{userId}/usrnme/{usrnme}/currentPass/{currentPass}")
```

Comment:
1. update API is used for all types of updates including firstname..... we cannot expect that a firstname update by back office user would need him to know the password of the user. This change cannot be accepted.
2. If the expectation is that an existing user is trying to modify the password of their own userId, then basic/oauth authentication itself would have validated the current password.... so no need to check again

Reply:
Hell @nikpawar  did you work on this?